### DISCUSSION TITLE
Upgrade development?  Issue after changing phones

### DISCUSSION THREAD

**Discussion by rlp847** [mikebrady/shairport-sync#1331] [Q&A]
I installed the development build for Airplay 2 capabilities about a month ago and it was working well with both my iphone 7 and an iphone XR in the same house, my trusty iphone 7 finally died (4 years 11 months) so i upgraded to an iphone 12, and now audio will not play.  airplay is acting strange, when i attempt to play it will just pause, it will then show it's playing and get to the end of a song and just stop, when usually it continues on to the next song.  I'm having a hard time figuring out what i need to do here, any help?

Editing to add, I have 6 throughout my home on different pi's, all were working before, no audio to any of them with my new phone. is it an ios 15.1 issue?

Editing to add more, so the girlfriend got an iphone 12 mini as well, hers had ios 15.0.1 on it and airplay still works for her, so it has something to do with ios 15.1 it seems.

**Comment by rlp847** [comment]
so i saw in another post you referenced 15.1 was working in updated versions, so i went through the upgrade process.  is there any step-by-step documentation on that, because i sure felt like an idiot stumbling through that.

**Comment by mikebrady** [comment]
Thanks for the post. Yes, you should update all the Shairport Sync (SPS) instances -- 15.1 did introduce at least one new feature that caused older versions of the `development` branch of SPS to stop working. 

Updating SPS is simple. If you still have the directory into which you cloned Shairport Sync, then `cd` into it, ensure you have checked out the `development` branch and do a `git pull`. Then do the `./autoconf`, `./configure...`, `make`, `make install` steps as documented in the BUILDAP2 and finally a `systemctl restart shairport-sync`. You don't need to reinstall or update `nqptp` at the moment, but probably in a week or two -- things are changing there too.

If you have discarded the Shairport Sync directory, simply follow the BUILDFORAP2 guide completely except rather than do the enable and start steps at the end do `systemctl restart shairport-sync`.
